crate/operator/create.py

```diff
@@ -963,6 +1031,56 @@ async def create_system_user(
         )
 
 
+def is_shared_resources_cluster(node_spec: Dict[str, Any]) -> bool:
+    try:
+        return (
+            (
+                node_spec["resources"].get("limits", {}).get("cpu")
+                and node_spec["resources"].get("requests", {}).get("cpu")
+                and node_spec["resources"].get("limits", {}).get("cpu")
+                != node_spec["resources"].get("requests", {}).get("cpu")
+            )
+            or False
+        ) or (
+            (
+                node_spec["resources"].get("limits", {}).get("memory")
+                and node_spec["resources"].get("requests", {}).get("memory")
+                and node_spec["resources"].get("limits", {}).get("memory")
+                != node_spec["resources"].get("requests", {}).get("memory")
+            )
+            or False
+        )
+    except KeyError:
+        return False
```

Comment:
Could this be simplified? Considering that for it to be a shared cluster, the requests must be set and they must be different from limits, we could perhaps do something like this:
```suggestion
    cpu_req = node_spec["resources"].get("requests", {}).get("cpu")
    cpu_limit = node_spec["resources"].get("limits", {}).get("cpu")
    mem_req = node_spec["resources"].get("requests", {}).get("memory")
    mem_limit = node_spec["resources"].get("limits", {}).get("memory")
    if not cpu_req or mem_req:
        return False
    return cpu_req != cpu_limit or mem_req != mem_limit
```

Maybe a bit more readable? But up to you!